pandas/io/formats/format.py

```diff
@@ -897,11 +897,19 @@ def get_col_type(dtype):
                     lev3 = [blank] * clevels
                 if name:
                     lev3.append(lev.name)
```

Comment:
I think it might be better to restructure this whole loop.

iterating like this:

```
for g, grp in table.groupby(level=0):
      # grp is a DataFrame
```

would be simpler to print out I think.

Reply:
You are correct, using df.groupby makes it simpler. Change is implemented and I will ping you once it is green.

Reply:
I just had to revert my change. My attempt using df.groupby() changed the order of some elements and it didn't format more complex indicies like dates correctly.

The current approach does not have any of these issues, so I think it is the best I can do right now.

Reply:
ok, neither of these should be true, come back when you can

Reply:
@jreback The change in the order of some elements is caused by a bug in df.groupby() that I just reported. #17537

Reply:
ok thanks

Reply:
@MxKy Hi, I've fixed your issue https://github.com/pandas-dev/pandas/issues/17537.
Nothing stands in your way!